icepyx/core/granules.py

```diff
@@ -1,18 +1,19 @@
 import datetime
-import requests
-import time
 import io
-import re
 import json
-import numpy as np
 import os
 import pprint
-from xml.etree import ElementTree as ET
+import re
+import time
 import zipfile
+from xml.etree import ElementTree as ET
+
+import numpy as np
+import requests
 
 import icepyx.core.APIformatting as apifmt
-from icepyx.core.auth import EarthdataAuthMixin
 import icepyx.core.exceptions
+from icepyx.core.auth import EarthdataAuthMixin
```

Comment:
So either the linter and I disagree on the order of the letters of the alphabet, or there's a rule that says anything with `from` comes after anything starting with `import`... is this a setting/preference or PEP rule I just never knew about?

Reply:
>So either the linter and I disagree on the order of the letters of the alphabet, or

:laughing: Hey, maybe the linter is just living its bliss! :woozy_face: :laughing: 

PEP8 doesn't specify how imports should be ordered within groups, IIRC. This is a formatting behavior from [the "isort" ruleset](https://pycqa.github.io/isort/docs/configuration/custom_sections_and_ordering.html#controlling-how-isort-sections-from-imports). In each import group, "from" imports are sub-grouped below "straight" imports. The behavior can be disabled, but I personally find it makes the imports much easier to read because there's less zigging and zagging :)

Reply:
> but I personally find it makes the imports much easier to read because there's less zigging and zagging :)

Totally makes sense - and of course I find it makes them harder to read because I have to read through more lists (and those lists aren't separated by newlines)!

This is probably one of those rock-paper-scissors decisions, yeah?

Reply:
I'm happy for you to decide, I'm the newbie around here :)

Reply:
If we change our mind later, it's a simple config change and reformat away! :rocket:

Reply:
Can we leave it as abc by name (regardless of "from" or "import" preceding) for now? Thanks!

> If we change our mind later, it's a simple config change and reformat away! 🚀

Awesome - that's super good to know.

Reply:
TODO: Force sort within sections https://pycqa.github.io/isort/docs/configuration/options.html#force-sort-within-sections

Reply:
3c22771